### ISSUE TITLE
Building prediction fails even when standing still (no room).

### ISSUE THREAD

**Issue by overvolting** [Unvanquished/Unvanquished#562]
Sometimes, I try to construct a nice turret. The planned structure is yellow, but I can't build because 'no room'.

Maybe it should be red colored if I can't build.

**Comment by Viech** [comment]
Is this a duplicate of #563 or something that happens while you stand?

**Comment by illwieckz** [comment]
There is a similar issue sometime: the planned building is yellow/green, but in fact there is not enough place to build (for example there is an egg too near). When you order the build, the structures start to build then explode itself immediately, losing both structure and build points (which is very frustrating).

**Comment by overvolting** [comment]
This happens when I stand still.

**Comment by Viech** [comment]
OK then, it's possible that this is indeed a bug I introduced lately.

**Comment by overvolting** [comment]
It's very rare, I can think of one map where it tends to happen, I'll post the name here.

**Comment by Viech** [comment]
In that case, a screenshot would be very helpful.

**Comment by overvolting** [comment]
Ok, it's on Yocto if you build the human base up on the side wall. I'll try to get a screenshot.

It can happen on other maps, but it's very rare so I can't remember. It happened to me on forlorn too once even on 0.36.

**Comment by overvolting** [comment]
Wasn't able to reproduce. Closing, I'll re-open with a screenshot if i see it again.

**Comment by overvolting** [comment]
re-opening
screenshot:  http://postimg.org/image/pnf09l689/

**Comment by illwieckz** [comment]
Oh yes, I know it, it's because the structure invade your personal space. No need to build against a wall, just too near of you.

```
         o                                  |
        -|-                                 | wall
        / \                                 |
floor ---------------------------------------
       \___/\_____________________________/\___/
        red  green                          red
       \_____/\___________________________/\___/
         not   buildable                    not
   buildable                                buildable
```

**Comment by Viech** [comment]
Merging #661.

**Comment by overvolting** [comment]
illwieckz, by green do you mean yellow?

**Comment by Viech** [comment]
The diffuse color used by the can-build shader is [36a100](http://www.colorhexa.com/36a100) in RGB hex, so it's technically a green (though it appears more yellow-ish where multiple surfaces are alpha-blended). :wink:

**Comment by illwieckz** [comment]
Yes it's [green, super green](https://www.youtube.com/watch?v=lFeLDc2CzOs) ! :grinning:

[ But yes, The object texture is green, but in certain circumstances the rendering of this green texture can form some yellow details in the visible image. The difference is what is referred to, the nature of the object or its perception… Very philosophical isn't it? :smirk: ]

**Comment by bmorel** [comment]
Could you elaborate more on those "some cases"? A way to reproduce would be nice. I'm removing the tag in the meantime, though.
